### DISCUSSION TITLE
Intercept Table Creation

### DISCUSSION THREAD

**Discussion by mtnyoni** [sqlalchemy/sqlalchemy#11634] [Usage Questions]
I have been using Django. I have been liking how they create permissions table for each model when the model is migrated. So, I want to try to implement the same thing using SQLAlchemy. I am not sure if I am doing the right thing, but I was wondering if it's possible to intercept SQLAlchemy creating tables using events. 

I am using SQLAlchemy  with FastAPI and I planning to also use Alembic for migrations. But for the time being I am worried about the first time when the tables are created.

**Comment by CaselIT** [comment]
Hi,

It's documented at https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.DDLEvents
